src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/mendrelation/PublicTransportMendRelationAction.java

```diff
@@ -1120,16 +1120,17 @@ boolean isRestricted(Way currentWay, Way previousWay, Node commonNode) {
             else if (rel.hasTag("type", "restriction") && rel.hasKey("restriction"))
                 return false;
             else {
-                boolean remove = true;
                 String routeValue = relation.get("route");
                 for (String s : restrictions) {
+                    if (s.length() <= 12)
+                        continue;
                     String sub = s.substring(12);
                     if (routeValue.equals(sub) && rel.hasTag("type", s))
-                        remove = false;
+                        return false;
                     else if (routeValue.equals(sub) && rel.hasKey("restriction:" + sub))
```

Comment:
This would be the second part:
```suggestion
                    else if (routeValue.equals(s) && rel.hasKey(s))
```

Reply:
@floscher Thanks for helping, there. I haven't done any professional work in Java since Java 8 was fresh, so I'm cleaning off some rust here.

I just pushed these suggestions in dfcf644 and they still work on the test suite that I built up.

Reply:
No worries, it wasn't bad or anything :wink:. Thanks again for the change, it's now merged and should roll out soon.